### ISSUE TITLE
NPE on RetryingStream.close()

### ISSUE THREAD

**Issue by ericrrath** [oracle/oci-java-sdk#299]
When using the new [DownloadManager](https://docs.oracle.com/en-us/iaas/tools/java/1.34.0/com/oracle/bmc/objectstorage/transfer/DownloadManager.html) (v1.34.0) to download objects from ObjectStore, `RetryingStream.close()` throws a `NullPointerException`:
```
Caused by: java.lang.NullPointerException: Cannot invoke "com.oracle.bmc.objectstorage.responses.GetObjectResponse.getInputStream()" because "this.response" is null
    at com.oracle.bmc.objectstorage.transfer.internal.download.RetryingStream.close(RetryingStream.java:75) ~[?:?]
...
```
I suspect the calling code invoked `close()` multiple times.  But the docs for [`Closeable.close()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Closeable.html) state:
> If the stream is already closed then invoking this method has no effect.

This suggests that implementations of `close()` should be idempotent; i.e. calling `close()` multiple times on the same instance of `RetryingStream` should return without throwing an NPE.

**Comment by mricken** [comment]
Hi @ericrrath , thank you for reporting this, and I'm sorry for the long delay. We will include the fix in the next one or two releases.

**Comment by mricken** [comment]
Fix is scheduled for release on May 4, 2021.

**Comment by y-chandra** [comment]
This has been fixed in the version `1.36.3` of the SDK
